realm/realm-library/src/main/java/io/realm/HandlerController.java

```diff
@@ -469,13 +466,11 @@ private void completedAsyncQueriesUpdate(QueryUpdateTask.Result result) {
                 query.notifyChangeListeners();
             }
 
-            // notify listeners only when we advanced
-            if (compare != 0) {
-                notifyGlobalListeners();
-                // notify RealmResults & RealmObject callbacks (type based notifications)
-                notifySyncRealmResultsCallbacks();
-                notifyRealmObjectCallbacks();
-            }
+            // We need to notify the rest of listeners, since the original REALM_CHANGE
+            // was delayed/swallowed in order to be able to update async queries
```

Comment:
@nhachicha Why did you remove the `if (compare != 0) {`? if `compare == 0`, `advanceRead` is not called, and no changes in this `SharedGroup` instance.